javascript/portfolio.js

```diff
@@ -14,17 +15,40 @@ function PortfolioObject(portData){
 //     portObjectArray.push(new PortfolioObject(portfolioData[i]));
 //   }
 // }
-
+// ports.renderToPage() = function(){
+//   portObjectArray.forEach(function(i) {
+//     $('#portfolio').append(i.toHtml());
+//   });
+// };
 PortfolioObject.prototype.toHtml = function(){
   var source = $('#portfolio-template').html();
   var template = Handlebars.compile(source);
   return template(this);
 };
 // portSetup();
-rawData.forEach(function(i) {
-  portObjectArray.push(new PortfolioObject(i));
-});
-portObjectArray.forEach(function(i) {
-  $('#portfolio').append(i.toHtml());
-});
+PortfolioObject.loadAll = function(rawData){
+  rawData.forEach(function(i) {
+    portObjectArray.push(new PortfolioObject(i));
+  });
+};
+PortfolioObject.fetchAll = function(){
+  if(localStorage.rawData){
+    console.log('took this path');
+    PortfolioObject.loadAll(JSON.parse(localStorage.rawData));
+    portObjectArray.forEach(function(i) {
+      $('#portfolio').append(i.toHtml());
+    });
+  } else {
+    $.getJSON('javascript/data.json').then(function(rawData){
+      console.log('test test test');
+      PortfolioObject.loadAll(rawData);
+      localStorage.rawData = JSON.stringify(rawData);
+      portObjectArray.forEach(function(i) {
```

Comment:
Since you have written this code twice, I would suggest making this its own method on the ports object.